### PR TITLE
Support rename=True in named tuple inference

### PR DESCRIPTION
**Pull request by rogalski**
Fixes PyCQA/pylint#1367

### CODE REVIEW

**Comment by rogalski** [comment]
Is there a change to having it merged before 1.5.0?

**Comment by rogalski** [comment]
@PCManticore @ceridwen @degustaf it would be nice to get any form of input before it'll be merged.

File: astroid/brain/brain_namedtuple_enum.py [hunk 1]
```diff
@@ -113,6 +115,17 @@ def infer_named_tuple(node, context=None):
     """Specific inference function for namedtuple Call node"""
     class_node, name, attributes = infer_func_form(node, nodes.Tuple._proxied,
                                                    context=context)
+
+    call_site = arguments.CallSite.from_call(node)
+    try:
+        func = next(extract_node('import collections; collections.namedtuple').infer())
+        rename = next(call_site.infer_argument(func, 'rename', context)).bool_value()
+    except (InferenceError, StopIteration):
```
Comment (PCManticore) [inline astroid/brain/brain_namedtuple_enum.py]: If it raises StopIteration, then it is a bug. Let's not mask it, so we could know what exactly fails.

File: astroid/brain/brain_namedtuple_enum.py [hunk 2]
```diff
@@ -142,6 +155,17 @@ def __getnewargs__(self):
     return iter([class_node])
 
 
+def _get_renamed_namedtuple_atributes(field_names):
+    names = list(field_names)
+    seen = set()
+    for i, name in enumerate(names):
```
Comment (PCManticore) [inline astroid/brain/brain_namedtuple_enum.py]: ``enumerate(field_names)``

**Pull request merged**